packages/react-composites/src/composites/ChatComposite/file-sharing/FileUpload.ts

```diff
@@ -96,41 +100,30 @@ export interface FileUploadManager {
 }
 
 /**
- * An internal interface used by the Chat Composite to drive the UI for file uploads.
- * @beta
+ * A wrapper object for a file that is being uploaded.
+ * Provides common functions for updating the upload progress, canceling an upload etc.
+ * @internal
  */
-export interface ObservableFileUpload extends FileUploadEventEmitter {
-  /**
-   * Unique identifier for the file upload.
-   */
-  id: string;
+export class FileUpload implements FileUploadManager, FileUploadEventEmitter {
+  private _emitter: EventEmitter;
+  public id: string;
+  public file?: File;
   /**
    * Filename to be displayed in the UI during file upload.
    */
-  fileName: string;
+  public fileName: string;
```

Comment:
instead of a property I think this should be a getter helper function
`filename(): string | undefined =>  data.file?.name || data.metadata?.name;`

as currently there are three different fields that will give you the filename (file.name, metadata.name and fileName). Alternative if we combine file and metadata into one field called uploadData then filename would be myUploadFile.uploadData.name

Reply:
Slight correction, `fileName` contains data from `file.name` OR `metadata.name` so there are 2 fields that give us filename.
Since `file.name` takes precedence over `metadata.name` initializing `fileName` with the correct data is sufficient.